library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -1929,6 +2028,9 @@ impl<T, A: Allocator> VecDeque<T, A> {
     /// assert_eq!(buf, [1, 2, 3, 4]);
     /// assert_eq!(buf2, []);
     /// ```
+    /// # Complexity
+    ///
+    /// Takes `O(n)` time, where `n` is the size of the other VecDeque.
```

Comment:
This feels potentially misleading. After all, we will call `self.reserve(...)`, so this might be copying both self and other's elements, not just one of them.